src/main/java/com/google/sps/ImageTextDectector.java

```diff
@@ -0,0 +1,128 @@
+ 
+    return createShoppingListQuery(annotation);
+  }
+
+
+  private Image shoppingImageInitializer(byte[] shoppingImageBytes)
+       throws PhotoShoppingException {
+    return PhotoShoppingUtil.getImageFrombytes(shoppingImageBytes);
+  }
+
+
+  /** Generates the request query to be sent to CloudVisionAPI client. */
+  // Keeping it public so that it could be tested from the unit tests
+  private List<AnnotateImageRequest> shoppingImageRequestGenerator(Image shoppingImage) {
+    List<AnnotateImageRequest> requests = new ArrayList<>();
+
+    AnnotateImageRequest request =
+        AnnotateImageRequest.newBuilder()
+            .addFeatures(Constants.TEXT_DETECTION_FEATURE)
+            .setImage(shoppingImage)
+            .build();
+
+    requests.add(request);
+    return requests;
+  }
+
+  /**
+   * Sends request to cloudVisionAPI. The Cloud Vision API scans the image and returns back the
+   * text, its position and properties as the response.
+   */
+  private BatchAnnotateImagesResponse cloudVisionAPIQuerier(List<AnnotateImageRequest> requests)
+      throws PhotoShoppingException {
+    ImageAnnotatorClient cloudVisionClient;
+    try {
+      cloudVisionClient = ImageAnnotatorClient.create();
+    } catch (IOException exception) {
+      throw new PhotoShoppingException(
+          "Failed to create cloudVisionClient\n" + exception.getMessage(), exception);
+    }
+    BatchAnnotateImagesResponse response = cloudVisionClient.batchAnnotateImages(requests);
+    cloudVisionClient.close();
+    return response;
+  }
+
+  /**
+   * Takes cloudVisionAPI's response and returns a list of annotations. ToDo : The
+   * positions from annotation will be used in sentence formation algorithm to separate
+   * individual queries from the shopping list.
+   */
+  private List<EntityAnnotation> parseAnnotateImageResponse(BatchAnnotateImagesResponse response)
+      throws PhotoShoppingException {
+    List<String> shoppingList = new ArrayList<>();
+    List<AnnotateImageResponse> responses = response.getResponsesList();
+    List<EntityAnnotation> annotation = new ArrayList<>();
+    for (AnnotateImageResponse identifiedText : responses) {
+      if (identifiedText.hasError()) {
+        throw new PhotoShoppingException(
+            "An error occurred while identifying the text from the image\n"
+                + identifiedText.getError().getMessage());
+      }
+      annotation = identifiedText.getTextAnnotationsList();
+    }
+    return annotation;
+  }
+
+  /** Creates query from the text detected by cloudVision API. */
+  // Keeping it public so that it could be tested from the unit tests
+  private String createShoppingListQuery(List<EntityAnnotation> annotation)
+      throws PhotoShoppingException {
+    // ToDo: Make an algorithm to create query sentences by separating out text returned by
+    // cloudVisionAPI
+    // to group shoppping items based on their position (y axis).
```

Comment:
I think the linter separated here the comment lines 
```suggestion
    // cloudVisionAPI to group shoppping items based on their position (y axis).
```